### PR TITLE
[slack] Refactor for better end-user usability

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Better error messages in message thread for end-user context
- Error message does not clutter chat window now since in message thread
- Fix Detect SQL SELECT for Slack users not Hue users by sending the copy/paste message but not the gist(requires Hue account)

## How was this patch tested?

- Invalid Link
Screenshot 2021-06-01 at 12 34 58 PM

- Mark not having hue account scenario
Screenshot 2021-06-01 at 12 35 51 PM

- gets no Hue account message in thread to de-clutter chat
Screenshot 2021-06-01 at 12 36 09 PM

### CODE REVIEW

**Comment by Harshg999** [comment]
- Query document not found from link

Screenshot 2021-06-01 at 12 44 02 PM

File: desktop/core/src/desktop/lib/botserver/api_tests.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def test_send_message(self):
       data = json.loads(response.content)
 
       assert_equal(200, response.status_code)
-      chat_postMessage.assert_called_with(channel='channel-1', text='@api_user: message with link', blocks=None)
+      chat_postMessage(channel='channel-1', text='@api_user: message with link', blocks=None, thread_ts=None)
```
Comment (romainr) [inline desktop/core/src/desktop/lib/botserver/api_tests.py:84]: What is it doing?

**Review by romainr — APPROVED** [review]

File: desktop/core/src/desktop/lib/botserver/api_tests.py (hunk 1 above)
Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/api_tests.py:84]: - These are the arguments of [chat.postMessage Slack API method](https://api.slack.com/methods/chat.postMessage)
blocks enable us to prettify messages by designing in blocks (I generally use [Block Kit builder](https://app.slack.com/block-kit-builder/T01PLLV8PSS#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Hey%20there,%20I'm%20your%20SQL%20Assistant!%20I'm%20here%20to%20assist%20users%20with%20their%20SQL%20queries%20and%20much%20more.%5CnHere%20are%20the%20few%20things%20I%20can%20help%20you%20with:%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Share%20query/gist%20links*%20in%20channel%20which%20unfurls%20in%20a%20rich%20preview,%20showing%20query%20details%20and%20result%20in%20message%20thread%20if%20available.%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Create%20a%20Gist%20of%20your%20query,%20select%20channel%20and%20*share%20directly%20from%20the%20Hue%20Editor*%20window.%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Detect%20SQL%20SELECT%20statements*%20in%20the%20channel%20and%20suggests%20a%20gist%20link%20for%20improved%20query%20discussions.%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Type%20%60@Hue%20queries%60%20if%20you%20want%20to%20explore%20and%20use%20the%20latest%20query%20bank.%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Read%20the%20*%3C%7Cuser%20guide%3E*%20for%20more%20information.%22%7D%7D%5D%7D) to visualize)
text is simple text message
thread_ts param helps in messaging in the reply thread

Comment (Harshg999) [reply desktop/core/src/desktop/lib/botserver/api_tests.py:84]: Oops, nice find! Will fix this

**Pull request merged**